src/core/renderers/webgl/managers/FilterManager.js

```diff
@@ -603,8 +644,40 @@ export default class FilterManager extends WebGLManager
     {
         const minWidth = renderTarget.size.width * renderTarget.resolution;
         const minHeight = renderTarget.size.height * renderTarget.resolution;
-        const key = ((minWidth & 0xFFFF) << 16) | (minHeight & 0xFFFF);
+
+        let key = screenKey;
+
+        if (minWidth !== this._screenWidth
+            || minHeight !== this._screenHeight)
+        {
+            key = ((minWidth & 0xFFFF) << 16) | (minHeight & 0xFFFF);
+        }
 
         this.pool[key].push(renderTarget);
     }
+
+    /**
+     * Called before the renderer starts rendering.
+     *
+     */
+    onPrerender()
+    {
+        if (this._screenWidth !== this.renderer.view.width
+            || this._screenHeight !== this.renderer.view.height)
+        {
+            this._screenWidth = this.renderer.view.width;
+            this._screenHeight = this.renderer.view.height;
+
+            const textures = this.pool[screenKey];
+
+            if (textures)
+            {
+                for (let j = 0; j < textures.length; j++)
+                {
+                    textures[j].destroy(true);
+                }
+            }
+            this.pool[screenKey] = [];
+        }
+    }
```

Comment:
Hey, instead of destroying here, we could resize render targets instead?

Reply:
Extra destroy is fine, I dont want to get a sudden memory increase from 640x480 to 2560x1280.